optuna_dashboard/_preference_setting.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from optuna.storages import BaseStorage
+
+from .preferential._study import PreferentialStudy
+
+
+if TYPE_CHECKING:
+    from typing import Literal
+
+    OUTPUT_COMPONENT_TYPE = Literal["Note", "Artifact"]
+
+_SYSTEM_ATTR_FEEDBACK_COMPONENT_TYPE = "preference:component_type"
+_SYSTEM_ATTR_FEEDBACK_ARTIFACT_KEY = "preference:component_artifact_key"
+
+
+def _register_output_component(
+    study_id: int,
+    storage: BaseStorage,
+    component_type: OUTPUT_COMPONENT_TYPE,
+    artifact_key: str | None = None,
+) -> None:
+    storage.set_study_system_attr(
+        study_id=study_id,
+        key=_SYSTEM_ATTR_FEEDBACK_COMPONENT_TYPE,
+        value=component_type,
+    )
+    if artifact_key is not None:
+        storage.set_study_system_attr(
+            study_id=study_id,
+            key=_SYSTEM_ATTR_FEEDBACK_ARTIFACT_KEY,
+            value=artifact_key,
+        )
+
+
+def register_output_component(
```

Comment:
This function is only for preferential module. So could you update the function name?